tests/__init__.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+
+from typing import Callable
+
+
+def check(actual: object, klass: type, dtype: type | None = None) -> None:
```

Comment:
One advantage of having separate `check` functions for each type (as was done prior to the introduction of `assert_type()`) is that we get extra type checking.  So `check_dataframe_result(result: pd.DataFrame)` is then checked by the type checker if we do `check_dataframe_result(pd.DataFrame({"x": [1,2,3]})`

Reply:
If the `check` functions becomes too large (maybe when checking the dtype), it would definitely be good to have dedicated check functions. At the moment I slightly prefer one function with a short name to keep the tests somewhat more skimmable (avoiding more line break).